collector/client.go

```diff
@@ -38,55 +28,36 @@ type SquidClient interface {
 	GetInfos() (types.Counters, error)
 }
 
-const (
-	requestProtocol = "GET cache_object://localhost/%s HTTP/1.0"
-)
-
-func buildBasicAuthString(login string, password string) string {
-	if len(login) == 0 {
-		return ""
-	} else {
-		return base64.StdEncoding.EncodeToString([]byte(login + ":" + password))
-	}
-}
-
-type CacheObjectRequest struct {
-	Hostname string
-	Port     int
-	Login    string
-	Password string
-	Headers  []string
-}
-
 /*NewCacheObjectClient initializes a new cache client */
-func NewCacheObjectClient(cor *CacheObjectRequest) *CacheObjectClient {
+func NewCacheObjectClient(hostname string, port int, username, password string, headers []string) *CacheObjectClient {
 	return &CacheObjectClient{
-		&connectionHandlerImpl{
-			cor.Hostname,
-			cor.Port,
-		},
-		buildBasicAuthString(cor.Login, cor.Password),
-		cor.Headers,
+		fmt.Sprintf("http://%s:%d/squid-internal-mgr/", hostname, port),
```

Comment:
is this backward compatible with older versions of squid?

Reply:
Thank you for confirming, 2011 sounds old enough to risk the backward compatibility issues. I'm wondering why didn't I use this in the first place :)